src/analysis/convex/exposed.lean

```diff
@@ -164,41 +175,15 @@ begin
   exact hC.inter_left hCA,
 end
 
-protected lemma is_extreme (hAB : is_exposed 𝕜 A B) :
-  is_extreme 𝕜 A B :=
-begin
-  refine ⟨hAB.subset, λ x₁ hx₁A x₂ hx₂A x hxB hx, _⟩,
-  obtain ⟨l, rfl⟩ := hAB ⟨x, hxB⟩,
-  have hl : convex_on 𝕜 univ l := l.to_linear_map.convex_on convex_univ,
-  have hlx₁ := hxB.2 x₁ hx₁A,
-  have hlx₂ := hxB.2 x₂ hx₂A,
-  refine ⟨⟨hx₁A, λ y hy, _⟩, ⟨hx₂A, λ y hy, _⟩⟩,
-  { rw hlx₁.antisymm (hl.le_left_of_right_le (mem_univ _) (mem_univ _) hx hlx₂),
-    exact hxB.2 y hy },
-  { rw hlx₂.antisymm (hl.le_right_of_left_le (mem_univ _) (mem_univ _) hx hlx₁),
-    exact hxB.2 y hy }
-end
-
-protected lemma convex (hAB : is_exposed 𝕜 A B) (hA : convex 𝕜 A) :
-  convex 𝕜 B :=
-begin
-  obtain rfl | hB := B.eq_empty_or_nonempty,
-  { exact convex_empty },
-  obtain ⟨l, rfl⟩ := hAB hB,
-  exact λ x₁ hx₁ x₂ hx₂ a b ha hb hab, ⟨hA hx₁.1 hx₂.1 ha hb hab, λ y hy,
-    ((l.to_linear_map.concave_on convex_univ).convex_ge _
-    ⟨mem_univ _, hx₁.2 y hy⟩ ⟨mem_univ _, hx₂.2 y hy⟩ ha hb hab).2⟩,
-end
-
-protected lemma is_closed [order_closed_topology 𝕜] (hAB : is_exposed 𝕜 A B) (hA : is_closed A) :
-  is_closed B :=
+protected lemma is_closed [nontrivial 𝕜] [order_closed_topology 𝕜] {A B : set E}
```

Comment:
I don't think you need nontrivial here: it only matters if `B` is empty, but then the empty set is always closed.